src/pt/blackoutcomics/src/eu/kanade/tachiyomi/extension/pt/blackoutcomics/BlackoutComics.kt

```diff
@@ -32,24 +42,74 @@ class BlackoutComics : ParsedHttpSource() {
 
     override val client by lazy {
         network.client.newBuilder()
-            .addInterceptor { chain ->
-                val response = chain.proceed(chain.request())
-                val request = response.request
-                if (request.url.pathSegments.contains("login")) {
-                    throw IOException("Faça o login na WebView para acessar o contéudo")
-                }
-                response
-            }
+            .addInterceptor(::checkingCredentials)
+            .addInterceptor(::authenticate)
             .rateLimitHost(baseUrl.toHttpUrl(), 2)
             .build()
     }
 
+    private val credentials: Pair<String, String> get() {
+        val username = preferences.getString(USERNAME_PREF, "")!!
+        val password = preferences.getString(PASSWORD_PREF, "")!!
+        return username to password
+    }
+
+    private fun checkingCredentials(chain: Interceptor.Chain): Response {
+        val (username, password) = credentials
+        if (username.isBlank() || password.isBlank()) {
+            throw IOException(
+                """
+                Configure suas credencias em Extensões > $name > Configuração.
+                """.trimIndent(),
+            )
+        }
+        return chain.proceed(chain.request())
+    }
+
+    private fun authenticate(chain: Interceptor.Chain): Response {
+        val response = chain.proceed(chain.request())
+        if (response.request.url.pathSegments.contains("temp")) {
+            return doAuth(response, chain)
+        }
+        return response
+    }
+
+    private fun doAuth(
+        response: Response,
+        chain: Interceptor.Chain,
+    ): Response {
+        val csrf = response.asJsoup()
+            .selectFirst("input[name='_token']")
+            ?.attr("value") ?: ""
+
+        val form = FormBody.Builder()
+            .add("_token", csrf)
+            .add("email", credentials.first)
+            .add("password", credentials.second)
+            .build()
+
+        chain.proceed(POST("$baseUrl/blackout/login", headers, form)).use {
+            if (it.request.url.pathSegments.contains("temp")) {
+                throw IOException(
+                    """
+                    Falha ao acessar recurso: Usuário ou senha incorreto.
+                    Altere suas credencias em Extensões > $name > Configuração.
+                    """.trimIndent(),
+                )
+            }
+        }
+        return chain.proceed(popularMangaRequest(0))
```

Comment:
```suggestion
        return chain.proceed(response.request)
```
try the previous request?

Reply:
> response.request

Previous request is login page. If you're not logged in, it redirects to the login page, which is why I forced popularMangaRequest.

Reply:
it redirects, so save the old request (chain.request() in line 70) and use that?